sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -199,23 +199,23 @@ case class Cast(child: Expression, dataType: DataType) extends UnaryExpression w
     // TimestampWritable.floatToTimestamp
     case FloatType =>
       buildCast[Float](_, f => try {
-        decimalToTimestamp(Decimal(f))
+        decimalToTimestamp(Decimal(f.toString))
```

Comment:
which test case is this? I think it is ok to lose precision if it is a float.

cc @marmbrus